app/services/bill-runs/two-part-tariff/submit-remove-bill-run-licence.service.js

```diff
@@ -0,0 +1,130 @@
+      .update('status', 'empty')
+      .from('billRuns')
+      .where('id', billRunId)
+
+    return true
+  }
+
+  // NOTE: The banner message is only set if licences remain in the bill run. This is because if there are no longer any
+  // licences remaining in the bill run the user is redirected to the "Bill runs" page instead of "Review licences". As
+  // the banner isn't displayed on the "Bill runs" page the message would remain in the cookie which could cause issues.
+  yar.flash('banner', `Licence ${licenceRef} removed from the bill run.`)
+
+  return false
+}
+
+async function _removeChargeElements (billRunId, licenceId) {
+  return db
+    .del()
+    .from('reviewChargeElements AS rce')
+    .innerJoin('reviewChargeReferences AS rcr', 'rce.reviewChargeReferenceId', 'rcr.id')
+    .innerJoin('reviewChargeVersions AS rcv', 'rcr.reviewChargeVersionId', 'rcv.id')
+    .innerJoin('reviewLicences AS rl', 'rcv.reviewLicenceId', 'rl.id')
+    .where('rl.billRunId', billRunId)
+    .andWhere('rl.licenceId', licenceId)
+}
+
+async function _removeChargeElementReturns (billRunId, licenceId) {
+  return db
+    .del()
+    .from('reviewChargeElementsReturns AS rcer')
+    .innerJoin('reviewChargeElements AS rce', 'rcer.reviewChargeElementId', 'rce.id')
+    .innerJoin('reviewChargeReferences AS rcr', 'rce.reviewChargeReferenceId', 'rcr.id')
+    .innerJoin('reviewChargeVersions AS rcv', 'rcr.reviewChargeVersionId', 'rcv.id')
+    .innerJoin('reviewLicences AS rl', 'rcv.reviewLicenceId', 'rl.id')
+    .where('rl.billRunId', billRunId)
+    .andWhere('rl.licenceId', licenceId)
+}
+
+async function _removeChargeReferences (billRunId, licenceId) {
+  return db
+    .del()
+    .from('reviewChargeReferences AS rcr')
+    .innerJoin('reviewChargeVersions AS rcv', 'rcr.reviewChargeVersionId', 'rcv.id')
+    .innerJoin('reviewLicences AS rl', 'rcv.reviewLicenceId', 'rl.id')
+    .where('rl.billRunId', billRunId)
+    .andWhere('rl.licenceId', licenceId)
+}
+
+async function _removeChargeVersions (billRunId, licenceId) {
+  return db
+    .del()
+    .from('reviewChargeVersions AS rcv')
+    .innerJoin('reviewLicences AS rl', 'rcv.reviewLicenceId', 'rl.id')
+    .where('rl.billRunId', billRunId)
+    .andWhere('rl.licenceId', licenceId)
+}
+
+async function _removeLicence (billRunId, licenceId) {
+  return db
+    .del()
+    .from('reviewLicences')
+    .where('billRunId', billRunId)
+    .andWhere('licenceId', licenceId)
+}
+
+async function _removeReturns (billRunId, licenceId) {
+  return db
+    .del()
+    .from('reviewReturns AS rr')
+    .innerJoin('reviewLicences AS rl', 'rr.reviewLicenceId', 'rl.id')
+    .where('rl.billRunId', billRunId)
+    .andWhere('rl.licenceId', licenceId)
+}
+
+async function _setSupBillingTrueAndReturnRef (licenceId) {
+  const licence = await db
+    .update('includeInSrocTptBilling', true)
+    .from('licences')
+    .where('id', licenceId)
+    .returning('licenceRef')
+
+  return licence[0].licenceRef
+}
```

Comment:
If we use the Objection model we can replace this with 1 line. I would also suggest a change to the function name

```suggestion
async function _flagForSupplementaryBilling (licenceId) {
  const licence = await LicenceModel.query().findById(licenceId).patch({ includeInSrocTptBilling: true })

  return licence.licenceRef
}
```

Reply:
I'll make this change to use objection. Just to note though that for it to return the `licenceRef` I'll need to add `.returning('licenceRef')` to the end of it otherwise it just returns the number of records updated. This also means that it is no longer a 1 liner as it goes over the line length.